From f7d6f66ef65f705829e3f73c49d1ade1cd037cc1 Mon Sep 17 00:00:00 2001
From: Vibhore Vardhan <vibhore@ti.com>
Date: Wed, 5 Feb 2025 14:27:54 -0600
Subject: [PATCH 03/68] TI: arm64: dts: ti: k3-am62a7-sk: enable uart as wakeup
 source

The main_uart0 (linux serial console) can be used as a wakeup source
using I/O Daisychain to trigger system wakeup from Low Power Modes
like Deep Sleep. Based on the patch for am625.

Signed-off-by: Vibhore Vardhan <vibhore@ti.com>
Signed-off-by: Kendall Willis <k-willis@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62a7-sk.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-am62a7-sk.dts b/arch/arm64/boot/dts/ti/k3-am62a7-sk.dts
index a6f0d87a50d8..c4be6bee5fe7 100644
--- a/arch/arm64/boot/dts/ti/k3-am62a7-sk.dts
+++ b/arch/arm64/boot/dts/ti/k3-am62a7-sk.dts
@@ -640,6 +640,9 @@ &main_uart0 {
 	status = "okay";
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_uart0_pins_default>;
+	interrupts-extended = <&gic500 GIC_SPI 178 IRQ_TYPE_LEVEL_HIGH>,
+			<&main_pmx0 0x1c8>; /* (D14) UART0_RXD PADCONFIG114 */
+	interrupt-names = "irq", "wakeup";
 };
 
 /* Main UART1 is used for TIFS firmware logs */
-- 
2.47.2

